### ISSUE TITLE
[`function-paren-newline`] Missing support for async arrow functions

### ISSUE THREAD

**Issue by mdziekon** [eslint/eslint#13728]
**Tell us about your environment**

* **ESLint Version:** 7.10.0
* **Node Version:** 14.7.0
* **npm Version:** 6.14.7

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

`default`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    rules: {
        'function-paren-newline': [
            'error',
            'multiline',
        ],
    },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const test = async (foo,
    bar) => {};
```

```bash
./node_modules/.bin/eslint src/test.js
```

**What did you expect to happen?**

ESLint should report the following problems:
```
  1:20  error  Expected newline after '('                             function-paren-newline
  2:8   error  Expected newline before ')'                            function-paren-newline
```

ESLint should be able to autofix this issue by inserting missing newlines.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

**Are you willing to submit a pull request to fix this bug?**

Yes

**Cause of the problem**

In this rule's `getParenTokens` function, when analysing [`ArrowFunctionExpression`](https://github.com/eslint/eslint/blob/master/lib/rules/function-paren-newline.js#L220-L233), the code wrongly assumes that any arrow function always starts with a parenthesis opening. Since the introduction of async / await syntax, this is no longer the case.

**Scope of the problem**

This issue affects all possible options, not only the `multiline` one as shown in my example.

**Comment by yeonjuan** [comment]
@mdziekon Hi :) Thanks for the report
It seems missing support for the async arrow function.

* [online-demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGZ1bmN0aW9uLXBhcmVuLW5ld2xpbmU6IFtcImVycm9yXCIsIFwibXVsdGlsaW5lXCJdICovXG5cbmZ1bmN0aW9uIGZ1bmMoYmFyLFx0Ly8gZXJyb3JcbiAgICAgICAgICAgICAgIGJheikge30gLy8gZXJyb3JcblxudmFyIGFycm93ID0gKGJhciwgLy8gZXJyb3JcbiAgICAgICAgICAgICBiYXopID0+IHt9ICAvLyBlcnJvclxuXG52YXIgYXN5bmNBcnJvdyA9IGFzeW5jIChiYXIsIC8vIG5vIGVycm9yXG4gICAgICAgICAgICAgICAgICBiYXopID0+IHt9IC8vIG5vIGVycm9yIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnsiYnJvd3NlciI6ZmFsc2V9fX0=)

```js
/* eslint function-paren-newline: ["error", "multiline"] */

function func(bar,	// error
               baz) {} // error

var arrow = (bar, // error
             baz) => {}  // error

var asyncArrow = async (bar, // no error
                  baz) => {} // no error

```
